packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -0,0 +1,259 @@
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        return padding.top;
+      case AxisDirection.right:
+        return padding.right;
+      case AxisDirection.down:
+        return padding.bottom;
+      case AxisDirection.left:
+        return padding.left;
+    }
+    return null;
+  }
+
+  /// The padding in the cross-axis direction on the side to the right when
+  /// facing away from the zero scroll offset in the scroll axis direction. (In
+  /// other words, for a vertical downwards-growing list, the padding on the
+  /// left.)
+  ///
+  /// Only valid after layout has started, since before layout the render object
+  /// doesn't know what direction it will be laid out in.
+  double get startPadding {
+    assert(constraints != null);
+    assert(constraints.axisDirection != null);
+    assert(constraints.growthDirection != null);
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        return padding.right;
+      case AxisDirection.right:
+        return padding.top;
+      case AxisDirection.down:
+        return padding.left;
+      case AxisDirection.left:
+        return padding.bottom;
+    }
+    return null;
+  }
+
+  /// The padding in the cross-axis direction on the side to the left when
+  /// facing away from the zero scroll offset in the scroll axis direction. (In
+  /// other words, for a vertical downwards-growing list, the padding on the
+  /// right.)
+  ///
+  /// Only valid after layout has started, since before layout the render object
+  /// doesn't know what direction it will be laid out in.
+  double get endPadding {
+    assert(constraints != null);
+    assert(constraints.axisDirection != null);
+    assert(constraints.growthDirection != null);
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        return padding.left;
+      case AxisDirection.right:
+        return padding.bottom;
+      case AxisDirection.down:
+        return padding.right;
+      case AxisDirection.left:
+        return padding.top;
+    }
+    return null;
+  }
+
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverPhysicalParentData)
+      child.parentData = new SliverPhysicalParentData();
+  }
+
+  @override
+  void performLayout() {
+    if (child == null) {
+      geometry = SliverGeometry.zero;
+      return;
+    }
+    child.layout(
+      constraints.copyWith(
+        scrollOffset: math.max(0.0, constraints.scrollOffset - beforePadding),
+        overlap: 0.0,
+        remainingPaintExtent: constraints.remainingPaintExtent - calculatePaintOffset(constraints, from: 0.0, to: beforePadding),
+        crossAxisExtent: constraints.crossAxisExtent - (startPadding + endPadding),
```

Comment:
It would be more efficient to compute `startPadding + endPadding` in one go as `crossAxisPadding`